### PR TITLE
Issue#2591no2

### PR DESCRIPTION
**Pull request by ArturGoldyn**
Use @runClassInSeparateProcess annotation on test class to allow running a test class with process isolation.

### CODE REVIEW

File: src/Util/PHP/Template/TestCaseMethod.tpl.dist [hunk 1]
```diff
@@ -47,7 +47,12 @@ function __phpunit_run_isolated_test()
     $result->beStrictAboutTodoAnnotatedTests({isStrictAboutTodoAnnotatedTests});
     $result->beStrictAboutResourceUsageDuringSmallTests({isStrictAboutResourceUsageDuringSmallTests});
 
-    $test = new {className}('{methodName}', unserialize('{data}'), '{dataName}');
```
Comment (sebastianbergmann) [inline src/Util/PHP/Template/TestCaseMethod.tpl.dist]: Would it make sense to use a new, separate template here? Maybe `TestCaseClass.tpl.dist`.

File: src/Framework/TestCase.php [hunk 2]
```diff
@@ -134,6 +134,13 @@
     protected $runTestInSeparateProcess;
 
     /**
+     * Whether or not this class is to be run in a separate PHP process.
+     *
+     * @var bool
+     */
+    protected $runClassInSeparateProcess;
```
Comment (sebastianbergmann) [inline src/Framework/TestCase.php]: Following the minimal visibility principle, this should be `private`.

File: src/Util/PHP/Template/TestCaseMethod.tpl.dist (hunk 1 above)
Comment (ArturGoldyn) [inline src/Util/PHP/Template/TestCaseMethod.tpl.dist]: Of course, I think so, especially that I've just made changes to TestCaseMethod file.

File: src/Framework/TestCase.php (hunk 2 above)
Comment (ArturGoldyn) [inline src/Framework/TestCase.php]: I'll try that.

**Comment by ArturGoldyn** [comment]
This fails now, probably due to change in sebastian/diff package from 1.4.1 to 1.4.2 [https://github.com/sebastianbergmann/diff/https://github.com/sebastianbergmann/diff/](url)

**Comment by ArturGoldyn** [comment]
https://github.com/sebastianbergmann/diff/ is probably the reason for failing tests.

**Comment by ArturGoldyn** [comment]
@sebastianbergmann template added, variable access changed, and all tests passing.

**Comment by sebastianbergmann** [comment]
Thanks!

**Comment by KintradimCrux** [comment]
In the current 6.2 release the method TestSuite::createTest() only calls $_test->setRunClassInSeparateProcess() for tests with a data provider.
Can you add this initialization for tests without a data provider, too?

**Comment by KintradimCrux** [comment]
And the @beforeClass, @afterClass, setUp() and tearDown() methods maybe should be called only in the isolated environment running the class.

**Comment by sebastianbergmann** [comment]
@KintradimCrux Can you please open tickets for these bugs? Thanks!

**Pull request merged**